sys/key_value.js

```diff
@@ -102,58 +72,53 @@ class KVBucket {
                 limit: 1
             }
         };
-        return hyper.get(storeReq).then(returnRevision(req));
+        return hyper.get(storeReq).then((dbResult) => {
+            if (dbResult.body && dbResult.body.items && dbResult.body.items.length) {
+                const row = dbResult.body.items[0];
+                return {
+                    status: 200,
+                    headers: row.headers,
+                    body: row.value
+                };
+            } else {
+                throw new HTTPError({
+                    status: 404,
+                    body: {
+                        type: 'not_found',
+                        uri: req.uri,
+                        method: req.method
+                    }
+                });
+            }
+        });
     }
 
     putRevision(hyper, req) {
-        const rp = req.params;
-        let tid = rp.tid && mwUtil.coerceTid(rp.tid, 'key_value');
-
-        if (!tid) {
-            tid = (mwUtil.parseETag(req.headers && req.headers.etag) || {}).tid;
-            tid = tid || uuid.now().toString();
-        }
-
         if (mwUtil.isNoStoreRequest(req)) {
-            return {
-                status: 202,
-                headers: {
-                    etag: req.headers && req.headers.etag || mwUtil.makeETag('0', tid)
-                }
-            };
+            return { status: 202 };
         }
 
+        const rp = req.params;
+        const tid = uuid.now().toString();
+
+        const headersToStore = {};
```

Comment:
`content-type` should be added straight away if present and then have it be overwritten if it's also present as `x-store-content-type`

Reply:
I actually am on the edge here. Why are we making an exception with the content-type again? I honestly don't quite see any benefit in having an exception for it, only confusion.